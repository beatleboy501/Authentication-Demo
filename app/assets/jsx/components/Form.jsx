import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Input from './Input.jsx';

const propTypes = {
  onSubmit: React.PropTypes.func,
  onInputChange: React.PropTypes.func,
  onPasswordChange: React.PropTypes.func,
  username: React.PropTypes.String,
  password: React.PropTypes.String,
  optionalTitle: React.PropTypes.String
};

export default class Form extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
        <div className="row">
          <div className="col-md-3"><h4>{this.props.optionalTitle}</h4></div>
          <div className="col-md-3">
            <form onSubmit={this.props.onSubmit}>
              <Input name="username" type="text" label="Username:" valChange={this.props.onInputChange} val={this.props.newUsername}/>
              <Input name="password" type="password" label="Password:" valChange={this.props.onPasswordChange} val={this.props.newPassword}/>
              <br/>
              <button type="submit">Submit</button>
            </form>
          </div>
          <div className="col-md-3"></div>
        </div>
    );
  }
}

Form.PropTypes = propTypes;
