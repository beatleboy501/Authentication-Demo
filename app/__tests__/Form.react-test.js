import React from 'react';
import Form from '../assets/jsx/components/Form.jsx';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

test('Form renders', () => {
  const component = renderer.create(<Form></Form>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('User Form', () => {
  const submitSpy = jest.fn();
  const inputChangeSpy = jest.fn();
  const passwordChangeSpy = jest.fn();
  const form = mount(
      <Form identifier="userForm" onSubmit={submitSpy}
            onInputChange={inputChangeSpy}
            onPasswordChange={passwordChangeSpy} username=''
            password=''></Form>
  );
  expect(form.props()).toEqual({"identifier": "userForm", "onInputChange": inputChangeSpy, "onPasswordChange": passwordChangeSpy, "onSubmit": submitSpy, "password": "", "username": ""})
  form.find("input").forEach((input) => {
    input.simulate("change", "new val");
  });
  expect(inputChangeSpy).toHaveBeenCalled();
  expect(passwordChangeSpy).toHaveBeenCalled();
  form.find('form').simulate('submit');
  expect(submitSpy).toHaveBeenCalled();
});
