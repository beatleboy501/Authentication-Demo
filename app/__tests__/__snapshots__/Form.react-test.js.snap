// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders 1`] = `
<div
  className="row"
>
  <div
    className="col-md-3"
  />
  <div
    className="col-md-3"
  >
    <form
      onSubmit={undefined}
    >
      <div>
        <label>
          Username:
        </label>
        <br />
        <input
          onChange={undefined}
          type="text"
          value={undefined}
        />
        <br />
      </div>
      <div>
        <label>
          Password:
        </label>
        <br />
        <input
          onChange={undefined}
          type="password"
          value={undefined}
        />
        <br />
      </div>
      <br />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
  <div
    className="col-md-3"
  />
</div>
`;
